### PR TITLE
Feature/ben 85

### PR DESCRIPTION
**Pull request by Underlyingglitch**
## 🎟️ Issue
Linear ticket BEN-85

## ✨ Description
Added User model
Implemented authentication
Added HashPassword helper
Fixed SQL syntax errors in QueryBuilder
Fixed error in connectionString

### CODE REVIEW

File: ExotischNederland/DAL/SQLDAL.cs [hunk 1]
```diff
@@ -19,6 +19,31 @@ public SQLDAL()
...
+                connection.Open();
+                using (SqlCommand command = new SqlCommand(query, connection))
+                {
+                    using (SqlDataReader reader = command.ExecuteReader())
+                    {
+                        if (reader.Read())
+                        {
+                            // Use the fillable array to only fill the properties that are allowed
+                            var values = new Dictionary<string, object>();
+                            for (int i = 0; i < reader.FieldCount; i++)
```
Comment (Nah-Nova) [inline ExotischNederland/DAL/SQLDAL.cs:37]: nitpick(non-blocking): maybe add a extra comment or 2 here, could be hard to understand for @paulib2000

File: ExotischNederland/Helpers.cs [hunk 2]
```diff
@@ -49,5 +50,24 @@ static string MenuSelect(Dictionary<string, string> items, bool indexes = false)
                 }
             }
         }
+
+        /// <summary>
+        /// Returns a SHA256 hash of the input password
+        /// </summary>
+        /// <param name="_password">Input password</param>
+        /// <returns></returns>
+        public static string HashPassword(string _password)
```
Comment (Nah-Nova) [inline ExotischNederland/Helpers.cs:59]: praise: nice work! we should use this for all helpers :)

**Review by Nah-Nova — APPROVED** [review]
Nitpick (non-blocking): The code is well-structured, and you’ve done a great job overall. The only suggestion I have is to add more comments throughout the code. This will make it easier for everyone to understand it and help our group better understand the logic during the assessment.

**Comment by Underlyingglitch** [comment]
I will add more comments later on. Thank you

**Pull request merged**